Python/bytecodes.c

```diff
@@ -3645,31 +4043,37 @@ dummy_func(
         }
 
          op(_CALL_METHOD_DESCRIPTOR_O, (callable, self_or_null, args[oparg] -- res)) {
+            PyObject *callable_o = PyStackRef_AsPyObjectBorrow(callable);
+
             int total_args = oparg;
-            if (self_or_null != NULL) {
+            if (!PyStackRef_IsNull(self_or_null)) {
                 args--;
                 total_args++;
             }
-            PyMethodDescrObject *method = (PyMethodDescrObject *)callable;
+
+            PyMethodDescrObject *method = (PyMethodDescrObject *)callable_o;
             DEOPT_IF(total_args != 2);
             DEOPT_IF(!Py_IS_TYPE(method, &PyMethodDescr_Type));
             PyMethodDef *meth = method->d_method;
             DEOPT_IF(meth->ml_flags != METH_O);
             // CPython promises to check all non-vectorcall function calls.
             DEOPT_IF(tstate->c_recursion_remaining <= 0);
-            PyObject *arg = args[1];
-            PyObject *self = args[0];
+            _PyStackRef arg_stackref = args[1];
+            _PyStackRef self_stackref = args[0];
+            PyObject *self = PyStackRef_AsPyObjectBorrow(self_stackref);
```

Comment:
Can you move these closer to where they are used, maybe even to within the call.

Reply:
The `_PyCFunction_TrampolineCall` is a wasm-only thing that shouldn't know about PyStackRef I think.